sql/core/src/main/scala/org/apache/spark/sql/GroupedData.scala

```diff
@@ -273,6 +280,60 @@ class GroupedData protected[sql](
   def sum(colNames: String*): DataFrame = {
     aggregateNumericColumns(colNames : _*)(Sum)
   }
+
+  /**
+    * (Scala-specific) Pivots a column of the current [[DataFrame]] and preform the specified
+    * aggregation.
+    * {{{
+    *   // Compute the sum of earnings for each year by course with each course as a separate column
+    *   df.groupBy($"year").pivot($"course", "dotNET", "Java").agg(sum($"earnings"))
+    *   // Or without specifying column values
+    *   df.groupBy($"year").pivot($"course").agg(sum($"earnings"))
+    * }}}
```

Comment:
Yeah. Let's add that. In our error message, we can ask users to change the value of that conf (btw, let's make sure the default value of that conf is large enough for common use cases).